usb_test.go

```diff
@@ -94,3 +94,45 @@ func TestOpenDeviceWithVIDPID(t *testing.T) {
 		}
 	}
 }
+
+func TestOpenDeviceWithFileDescriptor(t *testing.T) {
+	ctx := newContextWithImpl(newFakeLibusb())
+	defer ctx.Close()
+
+	// file descriptor is an index to the FakeDevices array
+	for _, d := range []struct {
+		vid, pid  ID
+		sysDevPtr uintptr
+	}{
+		{0x9999, 0x0001, 78},
+		{0x8888, 0x0002, 94},
+	} {
+		dev, err := ctx.OpenDeviceWithFileDescriptor(d.sysDevPtr)
+		if dev == nil {
+			t.Errorf("OpenDeviceWithFileDescriptor(%d): device == nil for a valid device", d.sysDevPtr)
```

Comment:
Fatalf.

Also add a check to see if the obtained device is the one you asked for (does VID/PID match).

Reply:
these two above (line 112 and 115) still need to be Fatalf.

With Fatal(f), the test is immediately interrupted. With Error(f), the test continues, but will report a failure at the end with all recorded errors. The principle is "return as much useful test results from a single run as possible". 

Hence:
* if err is not nil, dev is undefined and no further testing can be done (Fatal)
* if dev is nil, there's no device, testing properties of the device makes no sense (Fatal)
* if dev is not nil and err is nil, then dev may have useful properties. Checking these properties likely doesn't affect whether further testing can be done (Error), unless the code after that check has a hard dependency on the data that the previous check validated.

For future reference: if there's enough distinct check flows that each check forms a chain of dependencies, it may make sense to split individual checks into a subtest (t.Run).